federation/pkg/federation-controller/sync/controller_test.go

```diff
@@ -95,34 +95,45 @@ func TestSyncToClusters(t *testing.T) {
 }
 
 func TestClusterOperations(t *testing.T) {
-	adapter := &federatedtypes.SecretAdapter{}
-	obj := adapter.NewTestObject("foo")
-	differingObj := adapter.Copy(obj)
-	federatedtypes.SetAnnotation(adapter, differingObj, "foo", "bar")
+	defaultAdapter := &federatedtypes.SecretAdapter{}
+	scheduleAdapter := &fakeSchedulerAdapter{}
+	obj := defaultAdapter.NewTestObject("foo")
+	differingObj := defaultAdapter.Copy(obj)
+	federatedtypes.SetAnnotation(defaultAdapter, differingObj, "foo", "bar")
 
 	testCases := map[string]struct {
 		clusterObject pkgruntime.Object
 		expectedErr   bool
 		operationType util.FederatedOperationType
+		adapter       federatedtypes.FederatedTypeAdapter
 	}{
 		"Accessor error returned": {
 			expectedErr: true,
+			adapter:     defaultAdapter,
```

Comment:
Consider simplifying so that the defaultAdapter is automatically used if not supplied in the test case.  This comment won't be valid if you accept me suggestion to extract scheduling, since it could then be tested separately.